content/blog/CJ-3-1-roadmap.mdx

```diff
@@ -0,0 +1,202 @@
+
+For more details about the design of CheerpJ, please refer to [the deep dive](/blog/cheerpj-3-deep-dive) we published at the time of the initial CheerpJ 3.0 release. You can also read the [architecture page](https://cheerpj.com/docs/explanation/architecture) in the CheerpJ documentation.
+
+## The Roadmap for Modern Java in the browser
+
+![CheerpJ roadmap](./roadmapcj3.png)
+
+Our vision for CheerpJ does not stop with Java 8, and is not limited to its role as a tool to run legacy Java applications.
+
+We are finally ready to announce that plans are in place for CheerpJ to support more modern Java versions (and corresponding OpenJDK runtimes), with a long-term ambition to reach parity with latest LTS releases of Java.
+
+Work to support Java 11+ is already well underway and will be merged to nightly builds as soon as possible after this release. This will constitute the key novel element of the next major release of CheerpJ, 4.0, which we expect between March and April 2025, with Java 17 support following later in the year. Support for Java 21 will be added in 2026, although the timeline might become more aggressive depending on commercial demand. In the longer term we plan to maintain LTS parity in CheerpJ, to fulfil our goal of making any Java applications run unmodified in the browser.
+
+With these developments, we are on a course to realise the full ambition of CheerpJ: a full Java environment for modern browsers, centred around up-to-date builds of OpenJDK, and a JVM in WebAssembly.
+
+Java developers will be able to look at the browser as just another target to distribute applications or libraries on, enabling a new generation of Web-enabled Java clients.
+
+While all this sounds exciting, it’s not the end of the story. CheerpJ 4.0 and future releases will not only introduce support for modern Java, but also new functionality that will allow applications that are not written in pure Java to also run on the browser.
+
+### What about Java apps with JNI components?
+
+CheerpJ 3.1 can effectively run any Java application in the browser, but the reality is that many Java applications are not _actually_ 100% Java.
+
+It’s fairly common to find Java applications that make extensive use of JNI code to access libraries or system capabilities that are not available to Java otherwise. This has historically been a considerable real-world limitation of CheerpJ, and up to CheerpJ 3.1 we could only offer the use of [JavaScript natives](https://cheerpj.com/docs/guides/implementing-native-libraries) as a workaround.
+
+CheerpJ 4.0 will introduce support for WebAssembly JNI modules, providing a complete solution for all cases where the C/C++ JNI code is available.
+
+To understand the promise and scope of this new feature, let’s consider a popular use case of CheerpJ: running Minecraft on the browser.
+
+<video controls autoplay loop muted playsinline>
+	<source src="./minecraft_video.mp4" type="video/mp4" />
```

Comment:
Review the cropping, there are 2 or 3 white pixels on the right side that are very distracting

Reply:
@GabrielaReyna Can we recrop the video to remove the extra white pixels on the right side?